connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResourceTest.java

```diff
@@ -196,20 +210,62 @@ public void testCreateConnectorExists() throws Throwable {
         PowerMock.verifyAll();
     }
 
-    @Test(expected = BadRequestException.class)
-    public void testCreateConnectorWithASlashInItsName() throws Throwable {
-        String badConnectorName = CONNECTOR_NAME + "/" + "test";
+    @Test
+    public void testCreateConnectorNameTrimWhitespaces() throws Throwable {
+        // Clone CONNECTOR_CONFIG_WITHOUT_NAME Map, as createConnector changes it (puts the name in it) and this
+        // will affect later tests
+        Map<String, String> inputConfig = new HashMap<>();
+        inputConfig.putAll(CONNECTOR_CONFIG_WITHOUT_NAME);
+        final CreateConnectorRequest bodyIn = new CreateConnectorRequest(CONNECTOR_NAME_PADDING_WHITESPACES, inputConfig);
+        final CreateConnectorRequest bodyOut = new CreateConnectorRequest(CONNECTOR_NAME, CONNECTOR_CONFIG);
```

Comment:
doesn't seem this is necessary? couldn't we just pass the parameters here directly to `putConnectorConfig` and the `ConnectorInfo` constructor? same with `bodyOut` in tests below.

Reply:
Not sure I fully understand this comment tbh. Do you mean we just need testPutConnectorConfig and skip the extra tests that actually call createConnector?